repos/system_upgrade/el8toel9/actors/checkvdo/libraries/checkvdo.py

```diff
@@ -113,42 +135,41 @@ def _process_undetermined_conversion_devices(devices):
-        remedy_hint = ('Install the \'vdo\' package and re-run upgrade to '
-                       'check for VDO devices requiring conversion or confirm '
-                       'that all VDO devices, if any, are managed by LVM.')
+    if not unchecked:
+        return False
+
+    if api.current_actor().get_vdo_answer():
+        # User asserted no VDO devices are present
+        _report_skip_check()
+        return True
+
+    summary = (
+        'The check of block devices could not be performed as the \'vdo\' '
+        'package is not installed. All VDO devices must be converted to '
+        'LVM management prior to the upgrade to prevent the loss of data.')
+    remedy_hint = ('Install the \'vdo\' package and re-run upgrade to '
+                   'check for VDO devices requiring conversion or confirm '
+                   'that all VDO devices, if any, are managed by LVM.')
 
-        reporting.create_report([
-            reporting.Title('Cannot perform the VDO check of block devices'),
-            reporting.Summary(summary),
-            reporting.Severity(reporting.Severity.HIGH),
-            reporting.Groups([reporting.Groups.SERVICES, reporting.Groups.DRIVERS]),
-            reporting.Remediation(hint=remedy_hint),
-            reporting.Groups([reporting.Groups.INHIBITOR])
-        ])
+    reporting.create_report([
+        reporting.Title('Cannot perform the VDO check of block devices'),
+        reporting.Summary(summary),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Groups([reporting.Groups.SERVICES, reporting.Groups.DRIVERS]),
+        reporting.Remediation(hint=remedy_hint),
+        reporting.Groups([reporting.Groups.INHIBITOR])
+    ])
+    return True
 
 
 def check_vdo(conversion_info):
-    all_vdo_converted = api.current_actor().get_all_vdo_converted_response()
-    if all_vdo_converted:
-        summary = ('User has asserted all VDO devices on the system have been '
-                   'successfully converted to LVM management or no VDO '
-                   'devices are present.')
+    _process_pre_conversion_vdos(conversion_info.pre_conversion)
+    _process_post_conversion_vdos(conversion_info.post_conversion)
 
-        reporting.create_report([
-            reporting.Title('Skipping the VDO check of block devices'),
-            reporting.Summary(summary),
-            reporting.Severity(reporting.Severity.INFO),
-            reporting.Groups([reporting.Groups.SERVICES, reporting.Groups.DRIVERS]),
-        ])
-    else:
-        _process_pre_conversion_vdos(conversion_info.pre_conversion)
-        _process_post_conversion_vdos(conversion_info.post_conversion)
-        _process_undetermined_conversion_devices(conversion_info.undetermined_conversion)
-        _process_failed_check_devices(conversion_info)
+    detected = _process_undetermined_conversion_devices(conversion_info.undetermined_conversion)
+    detected |= _process_failed_check_devices(conversion_info)
```

Comment:
I'm not a fan of using bitwise operators with logical values.  It's something that technically works but mixes, imo, different concepts.
That having been said, if it's consistent with leapp coding guidelines so be it.

Reply:
Results split into separate variables.